stdlib/dynarray.mli

```
@@ -0,0 +1,418 @@
+   must be synchronized (for instance with a {!Mutex.t}).
+*)
+
+(** {1:dynarrays Dynamic arrays} *)
+
+type 'a t
+(** A dynamic array containing values of type ['a].
+
+    A dynamic array [a] is an array, that is, it provides
+    constant-time [get] and [set] operation on indices between [0] and
+    [Dynarray.length a - 1] included. Its {b length} may change over
+    time by adding or removing elements to the end of the array.
+*)
+
+val create : unit -> 'a t
+(** [create ()] is a new, empty array. *)
+
+val make : int -> 'a -> 'a t
+(** [make n x] makes a array of length [n], filled with [x]. *)
+
+val init : int -> (int -> 'a) -> 'a t
+(** [init n f] is a new array [a] of length [n],
+    such that [get a i] is [f i]. In other words,
+    [a] is the the array whose elements are
+    [f 0; f 1; f 2; ...; f (n - 1)].
+
+    This is similar to {!Array.init}. *)
+
+val get : 'a t -> int -> 'a
+(** [get a i] is the [i]-th element of [a], starting with index [0].
+
+    @raise Invalid_argument if the index is
+      invalid (not in [0 .. length a-1]). *)
+
+val set : 'a t -> int -> 'a -> unit
+(** [set a i x] sets the [i]-th element of [a] to be [x].
+
+    Just like {!get}, [i] must be between [0] and [length a - 1]
+    included. [set] does not add new elements to the array -- see
+    {!add_last} to add an element.
+
+    @raise Invalid_argument if the index is invalid. *)
+
+val length : _ t -> int
+(** [length a] is the number of elements in the array.
+    The last element of [a], if not empty, is [get a (length a - 1)].
+    This operation is constant time. *)
+
+val is_empty : 'a t -> bool
+(** [is_empty a] is [true] if [a] is empty, that is, if [length a = 0]. *)
+
+val copy : 'a t -> 'a t
+(** [copy a] is a shallow copy of [a], that can be modified independently. *)
+
+
+(** {1:adding Adding elements}
+
+    Note: all operations adding elements can raise [Failure] if the
+    length would need to grow beyond {!Sys.max_array_length}. *)
```

I dislike the use of modal verbs (`can`, `would`). Present tense works better IMHO. 
Also I don't think they should raise `Failure` I think they should raise `Invalid_argument` 

```
Note: all operations adding elements raise [Invalid_argument] if the
resulting length is greater than {!Sys.max_array_length}.
```